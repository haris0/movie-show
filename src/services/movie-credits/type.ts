export interface Cast {
  adult: boolean;
  gender: number;
  id: number;
  known_for_department: string;
  name: string;
  original_name: string;
  popularity: number;
  profile_path: null | string;
  cast_id?: number;
  character?: string;
  credit_id: string;
  order?: number;
  department?: string;
  job?: string;
  overview: string;
  release_date?: string;
  backdrop_path: null | string;
  genre_ids: number[];
  vote_count: number;
  original_language: 'en';
  original_title?: string;
  poster_path: null | string;
  title?: string;
  video?: boolean;
  vote_average: number;
  media_type: 'movie' | 'tv';
  first_air_date?: string;
  origin_country?: string[];
  episode_count?: number;
  release_year?: number | '';
}
