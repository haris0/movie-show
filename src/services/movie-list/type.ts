import type { Result } from "../../types/movie";

export type CategoryType = 'now_playing' | 'popular' | 'top_rated' | 'upcoming';

export interface MovieListParams {
  category?: CategoryType;
  page?: number;
}

export interface Movies {
  dates: Dates
  page: number
  results: Result[]
  total_pages: number
  total_results: number
}

export interface Dates {
  maximum: string
  minimum: string
}
