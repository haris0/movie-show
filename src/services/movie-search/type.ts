import type { Result } from "../../types/movie"

export interface SearchMovie {
  page: number
  results: Result[]
  total_pages: number
  total_results: number
}
