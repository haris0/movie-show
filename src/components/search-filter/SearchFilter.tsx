import type { CategoryType } from "../../services/movie-list/type";

interface SearchFilterProps {
  keyword?: string;
  setKeyword?: (keyword: string) => void;
  handleKeyDown?: React.KeyboardEventHandler<HTMLInputElement>;
  category: CategoryType;
  handleChangeCategory: (category: CategoryType) => void
}

const SearchFilter = ({
  keyword,
  setKeyword,
  handleKeyDown,
  category,
  handleChangeCategory
}: SearchFilterProps) => {
  return (
    <div className="w-full rounded-t-4xl -mt-8 min-h-20 bg-white flex justify-center">
      <div className="bg-white rounded-xl border border-slate-200 border-b-blue-100 p-5 w-full m-5 -mt-12 gap-4 flex flex-col">
        <div className="relative">
          <input
            name='search'
            type="text"
            className="bg-gray-50 p-2 pr-8 placeholder-gray-500 text-gray-500 text-sm rounded-lg w-full border border-slate-200 focus:outline-none" 
            placeholder="Search"
            value={keyword}
            onChange={(e) => setKeyword?.(e.target.value)}
            onKeyDown={handleKeyDown}
          />
          <svg 
            viewBox="0 0 24 24"
            className="h-5 w-5 ml-1 absolute top-2 right-2.5" 
            fill="none" 
            xmlns="http://www.w3.org/2000/svg"
          >
            <g id="SVGRepo_bgCarrier" strokeWidth="0"></g><g id="SVGRepo_tracerCarrier" strokeLinecap="round" strokeLinejoin="round"></g><g id="SVGRepo_iconCarrier"> <path d="M16.6725 16.6412L21 21M19 11C19 15.4183 15.4183 19 11 19C6.58172 19 3 15.4183 3 11C3 6.58172 6.58172 3 11 3C15.4183 3 19 6.58172 19 11Z" stroke="#737373" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"></path> </g>
          </svg>
        </div>
        <div className="relative">
          <select
            name='category'
            className="bg-gray-50 p-2 text-gray-500 text-sm rounded-lg w-full border border-slate-200 focus:outline-none appearance-none cursor-pointer"
            value={category}
            onChange={(e) => handleChangeCategory(e.target.value as CategoryType)}
          >
            <option value="now_playing">Now Playing</option>
            <option value="popular">Popular</option>
            <option value="top_rated">Top Rated</option>
            <option value="upcoming">Upcoming</option>
          </select>
          <svg 
            viewBox="0 0 24 24" 
            fill="none" 
            xmlns="http://www.w3.org/2000/svg"
            className="h-5 w-5 ml-1 absolute top-2.5 right-2.5"
          >
            <g id="SVGRepo_bgCarrier" strokeWidth="0"></g><g id="SVGRepo_tracerCarrier" strokeLinecap="round" strokeLinejoin="round"></g><g id="SVGRepo_iconCarrier"> <path d="M4 5L10 5M10 5C10 6.10457 10.8954 7 12 7C13.1046 7 14 6.10457 14 5M10 5C10 3.89543 10.8954 3 12 3C13.1046 3 14 3.89543 14 5M14 5L20 5M4 12L16 12M16 12C16 13.1046 16.8954 14 18 14C19.1046 14 20 13.1046 20 12C20 10.8954 19.1046 10 18 10C16.8954 10 16 10.8954 16 12ZM8 19L20 19M8 19C8 17.8954 7.10457 17 6 17C4.89543 17 4 17.8954 4 19C4 20.1046 4.89543 21 6 21C7.10457 21 8 20.1046 8 19Z" stroke="#737373" strokeWidth="1.5" strokeLinecap="round"></path> </g>
          </svg>
        </div>
      </div>
    </div>
  );
};

export default SearchFilter;
