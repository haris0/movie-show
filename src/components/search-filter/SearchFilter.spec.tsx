import { render } from '@testing-library/react';
import SearchFilter from './SearchFilter';
import type { CategoryType } from '../../services/movie-list/type';

const mockData = {
  keyword: '',
  setKeyword: () => undefined,
  handleKeyDown: () => undefined,
  category: 'now_playing' as CategoryType,
  handleChangeCategory: () => undefined,
};

describe('SearchFilter Component', () => {
  it('should render correctly', () => {
    const { baseElement } = render(
      <SearchFilter {...mockData}/>,
    );
    expect(baseElement).toMatchSnapshot();
  });
});
