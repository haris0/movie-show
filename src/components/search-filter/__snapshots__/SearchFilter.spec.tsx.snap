// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SearchFilter Component should render correctly 1`] = `
<body>
  <div>
    <div
      class="w-full rounded-t-4xl -mt-8 min-h-20 bg-white flex justify-center"
    >
      <div
        class="bg-white rounded-xl border border-slate-200 border-b-blue-100 p-5 w-full m-5 max-w-300 -mt-10 gap-4 flex flex-col"
      >
        <div
          class="relative"
        >
          <input
            class="bg-gray-50 p-2 pr-8 placeholder-gray-500 text-gray-500 text-sm rounded-lg w-full border border-slate-200 focus:outline-none"
            name="search"
            placeholder="Search"
            type="text"
            value=""
          />
          <svg
            class="h-5 w-5 ml-1 absolute top-2 right-2.5"
            fill="none"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="SVGRepo_bgCarrier"
              stroke-width="0"
            />
            <g
              id="SVGRepo_tracerCarrier"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
            <g
              id="SVGRepo_iconCarrier"
            >
               
              <path
                d="M16.6725 16.6412L21 21M19 11C19 15.4183 15.4183 19 11 19C6.58172 19 3 15.4183 3 11C3 6.58172 6.58172 3 11 3C15.4183 3 19 6.58172 19 11Z"
                stroke="#737373"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
               
            </g>
          </svg>
        </div>
        <div
          class="relative"
        >
          <select
            class="bg-gray-50 p-2 text-gray-500 text-sm rounded-lg w-full border border-slate-200 focus:outline-none appearance-none cursor-pointer"
            name="category"
          >
            <option
              value="now_playing"
            >
              Now Playing
            </option>
            <option
              value="popular"
            >
              Popular
            </option>
            <option
              value="top_rated"
            >
              Top Rated
            </option>
            <option
              value="upcoming"
            >
              Upcoming
            </option>
          </select>
          <svg
            class="h-5 w-5 ml-1 absolute top-2.5 right-2.5"
            fill="none"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="SVGRepo_bgCarrier"
              stroke-width="0"
            />
            <g
              id="SVGRepo_tracerCarrier"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
            <g
              id="SVGRepo_iconCarrier"
            >
               
              <path
                d="M4 5L10 5M10 5C10 6.10457 10.8954 7 12 7C13.1046 7 14 6.10457 14 5M10 5C10 3.89543 10.8954 3 12 3C13.1046 3 14 3.89543 14 5M14 5L20 5M4 12L16 12M16 12C16 13.1046 16.8954 14 18 14C19.1046 14 20 13.1046 20 12C20 10.8954 19.1046 10 18 10C16.8954 10 16 10.8954 16 12ZM8 19L20 19M8 19C8 17.8954 7.10457 17 6 17C4.89543 17 4 17.8954 4 19C4 20.1046 4.89543 21 6 21C7.10457 21 8 20.1046 8 19Z"
                stroke="#737373"
                stroke-linecap="round"
                stroke-width="1.5"
              />
               
            </g>
          </svg>
        </div>
      </div>
    </div>
  </div>
</body>
`;
